### PR TITLE
#7 Add dynamic navigation based on screen size

### LINKED ISSUE #7: Add Navigation Rail
Use Material design as per: https://developer.android.com/reference/kotlin/androidx/compose/material3/package-summary#navigationrail
Use Navigation Component: 

Tabs: Home, Camera, Profile - empty views

### PR DESCRIPTION
**Pull request by michal-kucznerowicz**
Add dynamic navigation depending on the current screen size based on Material 3 design guidelines. Of course, we can change the breakpoints of each type to suit our needs.

### CODE REVIEW

**Comment by BorysChajdas** [comment]
Would be better to separate the commits into two pull requests 😄 Changes regarding the .gitignore are not connected with this feature, so will be better to review them separately.

File: app/build.gradle [hunk 1]
```diff
@@ -52,12 +52,16 @@ dependencies {
...
 
-    implementation 'androidx.core:core-ktx:1.9.0'
-    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.5.1'
-    implementation 'androidx.activity:activity-compose:1.6.1'
     implementation 'androidx.compose.ui:ui'
     implementation 'androidx.compose.ui:ui-tooling-preview'
     implementation 'androidx.compose.material3:material3'
+    implementation 'androidx.compose.material:material-icons-extended'
+    implementation 'androidx.compose.material3:material3-window-size-class'
+
```
Comment (BorysChajdas) [inline app/build.gradle]: Why new line :D?

File: app/src/main/java/com/example/android/picover/MainActivity.kt [hunk 2]
```diff
@@ -1,40 +0,0 @@
-package com.example.android.picover
-
-import android.os.Bundle
-import androidx.activity.ComponentActivity
```
Comment (BorysChajdas) [inline app/src/main/java/com/example/android/picover/MainActivity.kt:4]: It will be better to move changing the package to another commit (and PR) to keep the history of `MainActivity`.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,3 +1,6 @@
 <resources>
     <string name="app_name">Picover</string>
+    <string name="home">Home</string>
```
Comment (BorysChajdas) [inline app/src/main/res/values/strings.xml:3]: Just to inform: I've added issue to discuss the naming convention for the strings: https://github.com/intive/Picover/issues/13

File: app/src/main/java/com/example/android/picover/presentation/navigation/NavigationType.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import androidx.compose.material3.windowsizeclass.WindowWidthSizeClass
+
+enum class NavigationType {
+    NAVIGATION_BAR,
+    NAVIGATION_RAIL,
+    NAVIGATION_DRAWER;
+
+    companion object {
+        fun fromWindowSize(windowSize: WindowSizeClass) = when (windowSize.widthSizeClass) {
+            WindowWidthSizeClass.Compact -> NAVIGATION_BAR
```
Comment (BorysChajdas) [inline app/src/main/java/com/example/android/picover/presentation/navigation/NavigationType.kt]: Redundant branch because `NAVIGATION_BAR` is in `else` block.

**Review by BorysChajdas — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (michal-kucznerowicz) [reply app/build.gradle]: To visually clarify which dependencies are automatically updated.

File: app/src/main/java/com/example/android/picover/MainActivity.kt (hunk 2 above)
Comment (michal-kucznerowicz) [reply app/src/main/java/com/example/android/picover/MainActivity.kt:4]: Done

File: app/src/main/java/com/example/android/picover/presentation/navigation/NavigationType.kt (hunk 4 above)
Comment (michal-kucznerowicz) [reply app/src/main/java/com/example/android/picover/presentation/navigation/NavigationType.kt]: Done

**Comment by michal-kucznerowicz** [comment]
> Would be better to separate the commits into two pull requests 😄 Changes regarding the .gitignore are not connected with this feature, so will be better to review them separately.

[https://github.com/intive/Picover/pull/15](url)

File: app/build.gradle (hunk 1 above)
Comment (marcinWisnia) [reply app/build.gradle]: is there any benefit of knowledge that the dependencies above the empty line are automatically updated and the rest not :D ?

File: app/src/main/java/com/example/android/picover/MainActivity.kt (hunk 2 above)
Comment (marcinWisnia) [reply app/src/main/java/com/example/android/picover/MainActivity.kt:4]: shouldn't @BorysChajdas resolve this conversation after checking if it was excluded to a separate MR :D ?

Comment (michal-kucznerowicz) [reply app/src/main/java/com/example/android/picover/MainActivity.kt:4]: Yes, but I don't remember how reviews work on GitHub 😄 that's why I resolved it. Additionally, we discussed offline that it's ok for now and we will work out some conventions at Thursday's meeting.

File: app/build.gradle (hunk 1 above)
Comment (michal-kucznerowicz) [reply app/build.gradle]: It seems not 😄. I can remove it.

Comment (michal-kucznerowicz) [reply app/build.gradle]: Done

File: app/build.gradle [hunk 5]
```diff
@@ -52,12 +52,15 @@ dependencies {
     implementation composeBom
     androidTestImplementation composeBom
 
-    implementation 'androidx.core:core-ktx:1.9.0'
-    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.5.1'
-    implementation 'androidx.activity:activity-compose:1.6.1'
     implementation 'androidx.compose.ui:ui'
     implementation 'androidx.compose.ui:ui-tooling-preview'
     implementation 'androidx.compose.material3:material3'
+    implementation 'androidx.compose.material:material-icons-extended'
```
Comment (BorysChajdas) [inline app/build.gradle:58]: If I remember correctly it is not recommended to add it because it is very heavy in terms of size :D It increases APK size a lot, but probably we can think about it later.

File: app/src/main/java/com/example/android/picover/presentation/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,77 @@
+package com.example.android.picover.presentation
```
Comment (BorysChajdas) [inline app/src/main/java/com/example/android/picover/presentation/MainActivity.kt]: There is still a problem with `MainActivity` because it is shown as deleted and added a new file :D. As I said in another comment – probably the change of package should be done in another commit (and another pull request).

**Review by BorysChajdas — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android/picover/MainActivity.kt (hunk 2 above)
Comment (BorysChajdas) [reply app/src/main/java/com/example/android/picover/MainActivity.kt:4]: I think it is still not resolved :D because MainActivity is showing as deleted

File: app/src/main/java/com/example/android/picover/presentation/MainActivity.kt (hunk 6 above)
Comment (michal-kucznerowicz) [reply app/src/main/java/com/example/android/picover/presentation/MainActivity.kt]: I think now it's done. Also, I changed PR to have only one commit and I will create a new PR with moving `MainActivity`.

File: app/build.gradle (hunk 5 above)
Comment (michal-kucznerowicz) [reply app/build.gradle:58]: You're right. It is said at the maven repository https://mvnrepository.com/artifact/androidx.compose.material/material-icons-extended. If we're only going to use a few of them I think a better way would be to add drawables to our resources for only those icons that we need.

**Review by BorysChajdas — APPROVED** [review]

**Review by marcinWisnia — APPROVED** [review]

**Review by bartek977 — APPROVED** [review]

**Comment by BorysChajdas** [comment]
@michal-kucznerowicz I will merge it as it has 3 approves 😄
